astropy/utils/compat/optional_deps.py

```diff
@@ -44,10 +44,6 @@
 
 def __getattr__(name):
     if name in __all__:
-        try:
-            importlib.import_module(_deps[name[4:]])
-        except (ImportError, ModuleNotFoundError):
-            return False
-        return True
+        return find_spec(name.removeprefix("HAS_").lower()) is not None
```

Comment:
This should be `find_spec(_deps[name[4:]])`

Reply:
at the risk of bike shedding, isn't `.removeprefix("HAS_")` clearer than `[4:]` (albeit ever so slightly slower) ?

Reply:
(though getting the name for `_deps` is clearly required, so I applied that fix)

Reply:
Yes, I agree, more obvious what it does.